src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -302,6 +303,32 @@ protected override void InvokeCallback()
             }
         }
 
+        private sealed class SendFileOperation : AsyncOperation
```

Comment:
@pgavlin, would you mind taking a look at this part to make sure it fits in correctly with your design?